other: 0.098
vnc: 0.096
KVM: 0.093
PID: 0.078
boot: 0.075
permissions: 0.071
graphic: 0.070
files: 0.069
semantic: 0.063
device: 0.062
socket: 0.061
performance: 0.061
network: 0.054
debug: 0.051
boot: 0.470
KVM: 0.386
PID: 0.037
debug: 0.033
files: 0.023
other: 0.011
device: 0.007
performance: 0.007
semantic: 0.006
network: 0.006
socket: 0.005
graphic: 0.004
vnc: 0.004
permissions: 0.003

bios_linker_loader_add_checksum: Assertion `start_offset < file->blob->len' failed

Trying boot/start a Windows 10 VM.  Worked until recently when this error started showing up.

I have the following installed on Fedora 33:
qemu-kvm-5.1.0-9.fc33.x86_64

This is the error:

Error starting domain: internal error: process exited while connecting to monitor: qemu-system-x86_64: /builddir/build/BUILD/qemu-5.1.0/hw/acpi/bios-linker-loader.c:239: bios_linker_loader_add_checksum: Assertion `start_offset < file->blob->len' failed.

Traceback (most recent call last):
  File "/usr/share/virt-manager/virtManager/asyncjob.py", line 65, in cb_wrapper
    callback(asyncjob, *args, **kwargs)
  File "/usr/share/virt-manager/virtManager/asyncjob.py", line 101, in tmpcb
    callback(*args, **kwargs)
  File "/usr/share/virt-manager/virtManager/object/libvirtobject.py", line 57, in newfn
    ret = fn(self, *args, **kwargs)
  File "/usr/share/virt-manager/virtManager/object/domain.py", line 1329, in startup
    self._backend.create()
  File "/usr/lib64/python3.9/site-packages/libvirt.py", line 1234, in create
    if ret == -1: raise libvirtError ('virDomainCreate() failed', dom=self)
libvirt.libvirtError: internal error: process exited while connecting to monitor: qemu-system-x86_64: /builddir/build/BUILD/qemu-5.1.0/hw/acpi/bios-linker-loader.c:239: bios_linker_loader_add_checksum: Assertion `start_offset < file->blob->len' failed.

I see this were referenced in a patch from some time ago and supposedly fixed.  Here is the patch info I was able to find:

http://next.<email address hidden><email address hidden>/

On Mon, 12 Apr 2021 20:29:04 -0000
Ed Davison <email address hidden> wrote:

> Public bug reported:
> 
> Trying boot/start a Windows 10 VM.  Worked until recently when this
> error started showing up.
> 
> I have the following installed on Fedora 33:
> qemu-kvm-5.1.0-9.fc33.x86_64

Could you add used QEMU command line in your case?

> 
> This is the error:
> 
> Error starting domain: internal error: process exited while connecting
> to monitor: qemu-system-x86_64: /builddir/build/BUILD/qemu-5.1.0/hw/acpi
> /bios-linker-loader.c:239: bios_linker_loader_add_checksum: Assertion
> `start_offset < file->blob->len' failed.
> 
> Traceback (most recent call last):
>   File "/usr/share/virt-manager/virtManager/asyncjob.py", line 65, in cb_wrapper
>     callback(asyncjob, *args, **kwargs)
>   File "/usr/share/virt-manager/virtManager/asyncjob.py", line 101, in tmpcb
>     callback(*args, **kwargs)
>   File "/usr/share/virt-manager/virtManager/object/libvirtobject.py", line 57, in newfn
>     ret = fn(self, *args, **kwargs)
>   File "/usr/share/virt-manager/virtManager/object/domain.py", line 1329, in startup
>     self._backend.create()
>   File "/usr/lib64/python3.9/site-packages/libvirt.py", line 1234, in create
>     if ret == -1: raise libvirtError ('virDomainCreate() failed', dom=self)
> libvirt.libvirtError: internal error: process exited while connecting to monitor: qemu-system-x86_64: /builddir/build/BUILD/qemu-5.1.0/hw/acpi/bios-linker-loader.c:239: bios_linker_loader_add_checksum: Assertion `start_offset < file->blob->len' failed.
> 
> I see this were referenced in a patch from some time ago and supposedly
> fixed.  Here is the patch info I was able to find:
> 
> http://next.patchew.org/QEMU/1515677902-23436-1-git-send-email-
> <email address hidden>/1515677902-23436-10-git-send-email-
> <email address hidden>/
> 
> ** Affects: qemu
>      Importance: Undecided
>          Status: New
> 



Hmmm.  Well, I don't know what the command line was.  I use Virtual Machine Manager (virt-manager.org) for my interface to the VM and it does the startup.  The error shows up when I start the VM.


On Tue, 13 Apr 2021 21:29:45 -0000
Ed Davison <email address hidden> wrote:

> Hmmm.  Well, I don't know what the command line was.  I use Virtual
> Machine Manager (virt-manager.org) for my interface to the VM and it
> does the startup.  The error shows up when I start the VM.
In this case you should be able to attach domain xml. (View->Details->Overview->XML)

Also try and see if the following patch helps:
https://<email address hidden>/T/#md70161e63276e9d5b6fd50fd835d2e62895810b8



The patch may be a bit beyond me at the moment as I use a package to install this and would have to figure out how to download source, get it configure, patched and compiled.  Whew!  Maybe ... 

But here is my XML config file.

On Wed, 14 Apr 2021 19:40:36 -0000
Ed Davison <email address hidden> wrote:

> The patch may be a bit beyond me at the moment as I use a package to
> install this and would have to figure out how to download source, get it
> configure, patched and compiled.  Whew!  Maybe ...
> 
> But here is my XML config file.
> 
> ** Attachment added: "domain xml file"
>    https://bugs.launchpad.net/qemu/+bug/1923497/+attachment/5487970/+files/win10-virt-domain.xml
> 

I don't see anything in this config that could trigger the assert.
(RAM size is 2Kb off 4Gb, but that's probably not the issue)

Can you provide a stack trace, it should help to find out
which path triggers assert.



Sorry, this has not recurred since you asked for the stack trace. Not sure what "fixed" it.  Closing the bug.

I guess I should say, you may close the bug. 

This was the fix:
https://gitlab.com/qemu-project/qemu/-/commit/bb9feea43179ef8aba2

